<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>666</width>
    <height>534</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>水声脉冲信号仿真界面</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QWidget" name="widget" native="true">
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string/>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="title">
              <property name="text">
               <string>水声脉冲信号仿真软件</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>345</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QPushButton" name="minButton">
                <property name="text">
                 <string>min</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="closeButton">
                <property name="text">
                 <string>close</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="maxButton">
                <property name="text">
                 <string>max</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="2,3">
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string/>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QGroupBox" name="groupBox_4">
                <property name="title">
                 <string>SIGNAL</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_4">
                 <item>
                  <widget class="QGroupBox" name="groupBox_9">
                   <property name="title">
                    <string/>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_7">
                    <item>
                     <widget class="QLabel" name="signalfs">
                      <property name="text">
                       <string>fs/Hz</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="signalfsSpinBox">
                      <property name="maximum">
                       <number>100000</number>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="signalT">
                      <property name="text">
                       <string>T/s</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QDoubleSpinBox" name="signalTSpinBox">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="signalNP">
                      <property name="text">
                       <string>NP</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="signalNPSpinBox"/>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <widget class="QGroupBox" name="groupBox_10">
                   <property name="title">
                    <string/>
                   </property>
                   <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1,1,1,1,0">
                    <item row="3" column="3">
                     <widget class="QDoubleSpinBox" name="sig3t0">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="4" colspan="2">
                     <widget class="QDoubleSpinBox" name="sig1t1">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item row="2" column="4" colspan="2">
                     <widget class="QDoubleSpinBox" name="sig2t1">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item row="2" column="0">
                     <widget class="QComboBox" name="sig2">
                      <item>
                       <property name="text">
                        <string>NO</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>CW</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>LFM</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>HFM</string>
                       </property>
                      </item>
                     </widget>
                    </item>
                    <item row="1" column="1">
                     <widget class="QSpinBox" name="sig1f0">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item row="2" column="2">
                     <widget class="QSpinBox" name="sig2f1">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="1">
                     <widget class="QLabel" name="label_3">
                      <property name="text">
                       <string>f0/Hz</string>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="3">
                     <widget class="QDoubleSpinBox" name="sig1t0">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item row="2" column="3">
                     <widget class="QDoubleSpinBox" name="sig2t0">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="4">
                     <widget class="QLabel" name="label_6">
                      <property name="text">
                       <string>t1/s</string>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="0">
                     <widget class="QLabel" name="label">
                      <property name="text">
                       <string>signal</string>
                      </property>
                     </widget>
                    </item>
                    <item row="3" column="2">
                     <widget class="QSpinBox" name="sig3f1">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="2">
                     <widget class="QSpinBox" name="sig1f1">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item row="3" column="1">
                     <widget class="QSpinBox" name="sig3f0">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="0">
                     <widget class="QComboBox" name="sig1">
                      <item>
                       <property name="text">
                        <string>NO</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>CW</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>LFM</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>HFM</string>
                       </property>
                      </item>
                     </widget>
                    </item>
                    <item row="3" column="4" colspan="2">
                     <widget class="QDoubleSpinBox" name="sig3t1">
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="2">
                     <widget class="QLabel" name="label_4">
                      <property name="text">
                       <string>f1/Hz</string>
                      </property>
                     </widget>
                    </item>
                    <item row="3" column="0">
                     <widget class="QComboBox" name="sig3">
                      <item>
                       <property name="text">
                        <string>NO</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>CW</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>LFM</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>HFM</string>
                       </property>
                      </item>
                     </widget>
                    </item>
                    <item row="2" column="1">
                     <widget class="QSpinBox" name="sig2f0">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="3">
                     <widget class="QLabel" name="label_5">
                      <property name="text">
                       <string>t0/s</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_7">
                <property name="title">
                 <string>WAV</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_2">
                 <item>
                  <widget class="QGroupBox" name="groupBox_8">
                   <property name="title">
                    <string/>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,5">
                    <property name="spacing">
                     <number>6</number>
                    </property>
                    <property name="leftMargin">
                     <number>4</number>
                    </property>
                    <property name="topMargin">
                     <number>4</number>
                    </property>
                    <property name="rightMargin">
                     <number>0</number>
                    </property>
                    <property name="bottomMargin">
                     <number>4</number>
                    </property>
                    <item>
                     <widget class="QPushButton" name="wavchoose">
                      <property name="text">
                       <string>wavpath</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="wavfilepath">
                      <property name="text">
                       <string/>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <widget class="QGroupBox" name="groupBox_5">
                   <property name="title">
                    <string/>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_6">
                    <item>
                     <widget class="QLabel" name="wavduration">
                      <property name="text">
                       <string>duration/s</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QDoubleSpinBox" name="wavdurationSpinBox">
                      <property name="value">
                       <double>5.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="wavstarttime">
                      <property name="text">
                       <string>starttime/s</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QDoubleSpinBox" name="wavstarttimeSpinBox"/>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_6">
                <property name="title">
                 <string>STFT</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <widget class="QGroupBox" name="groupBox_11">
                   <property name="title">
                    <string/>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_8">
                    <item>
                     <widget class="QLabel" name="snr">
                      <property name="text">
                       <string>snr</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="snrSpinBox">
                      <property name="minimum">
                       <number>-99</number>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="stftfs">
                      <property name="text">
                       <string>fs/Hz</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="stftfsSpinBox">
                      <property name="maximum">
                       <number>100000</number>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <widget class="QGroupBox" name="groupBox_12">
                   <property name="title">
                    <string/>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="1,1,1,1">
                    <item>
                     <widget class="QLabel" name="wlen">
                      <property name="text">
                       <string>wlen</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="wlenspinBox">
                      <property name="maximum">
                       <number>10000</number>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <spacer name="horizontalSpacer_4">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>49</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                    <item>
                     <widget class="QComboBox" name="stftcomboBox">
                      <item>
                       <property name="text">
                        <string>noisy_sig</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>noise</string>
                       </property>
                      </item>
                      <item>
                       <property name="text">
                        <string>sig</string>
                       </property>
                      </item>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_2">
             <property name="title">
              <string/>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout" stretch="1,8,1">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,3,1">
                <item>
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>View</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QComboBox" name="plotcomboBox">
                  <item>
                   <property name="text">
                    <string>stft</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>fft</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>wav</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QGraphicsView" name="stft_show"/>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QPushButton" name="start">
                  <property name="text">
                   <string>start</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="stop">
                  <property name="text">
                   <string>stop</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="save">
                  <property name="text">
                   <string>save</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
